doc/man/man3/crypto_curve_to_hidden.3monocypher

```diff
@@ -0,0 +1,283 @@
+indistinguishability of all communications from random noise,
+consider
+.Xr crypto_key_exchange 3monocypher
+instead.
+.Pp
+For understanding what these functions do, it is important to note that
+a
+.Dq public key
+in this context refers to a
+.Em point on Curve25519 .
+This also means that these functions yield data unsuitable for use with
+.Xr crypto_sign 3monocypher
+and related functions.
+.Pp
+.Fn crypto_curve_to_hidden
+takes a public key
+.Fa curve
+and a
+.Fa tweak ,
+hiding the public key it so that it is effectively indistinguishable
+from random noise.
+Note that only
+.Xr crypto_x25519_dangerous_fast 3monocypher
+or
+.Xr crypto_x25519_dangerous_small 3monocypher
+to generate a suitable public key;
+the
+.Xr crypto_x25519 3monocypher
+function is insufficient.
+The
+.Fa tweak
+must be chosen at random.
+Even then, this operation
+.Em may
+fail:
+Not all curve points are capable of being hidden.
+In this case,
+.Fn crypto_curve_to_hidden
+must be tried again with a new key pair.
+On average, two attempts are needed.
+Once a suitable public key has been found,
+.Fn crypto_curve_to_hidden
+always succeeds it.
+Given the same values for
+.Fa tweak
+and
+.Fa curve ,
+.Fn crypto_curve_to_hidden
+yields the same output value
+.Fa hidden .
+.Pp
+.Fn crypto_hidden_to_curve
+performs the inverse operation:
+It decodes a hidden point to a curve point on Curve25519.
+.Pp
+.Fn crypto_hidden_key_pair
+is a convenience function that generates a secret key and its
+corresponding public key, which is effectively indistinguishable from
+random noise from a random seed.
+The
+.Em execution time of this function is unpredictable
+because it may take many failures until a key pair could be generated
+successfully.
+.Fn crypto_hidden_key_pair
+uses
+.Xr crypto_x25519_dangerous_fast 3monocypher
+internally;
+if code size is an important concern,
+its functionality can be replicated with
+.Xr crypto_x25519_dangerous_fast 3monocypher
+instead.
+.Pp
+The arguments are:
+.Bl -tag -width Ds
+.It Fa curve
+A point on the curve, which is a Curve25519 public key generated with
+either
+.Xr crypto_x25519_dangerous_fast 3monocypher
+or
+.Xr crypto_x25519_dangerous_small 3monocypher .
+.It Fa hidden
+The hidden encoding of a point on the curve which is effectively
+indistinguishable from random.
+.It Fa secret_key
+The secret key that was generated from the given
+.Fa seed .
+.It Fa seed
+A 32-byte random number from which to derive a key pair.
+See
+.Xr intro 3monocypher
+for advice about generating random bytes (use the operating system's
+random number generator).
+The
+.Fa seed
+is wiped automatically.
+.It Fa tweak
+A 1-byte random number,
+which influences the final output of
+.Fn crypto_curve_to_hidden .
+.El
+.Pp
+The
+.Fa hidden ,
+.Fa curve ,
+and
+.Fa tweak
+arguments may overlap or point at the same buffer.
+.Sh RETURN VALUES
+.Fn crypto_curve_to_hidden
+returns 0 on success, -1 if the given
+.Fa curve
+argument is unsuitable for hiding.
```

Comment:
Not sure we should document it, but in addition, this function has no effect if it fails. (I should test this).